File: src/week3/P007.js
const fs = require('fs');
const [m, n] = fs.readFileSync('/dev/stdin').toString().trim().split(' ').map(Number);

// 소수 판별 함수
function isPrime(num) {
    if (num < 2) { // 1은 소수가 아님
        return false;
    }
    // 2부터 해당 수의 제곱근까지 반복
    for (let i = 2; i * i <= num; i++) {
        if (num % i === 0) {
            return false; // 나누어 떨어지면 소수가 아님
        }
    }
    return true;
}

const results = [];
for (let i = m; i <= n; i++) {
    if (isPrime(i)) {
        results.push(i);
    }
}

console.log(results.join('\n'));

File: src/week3/P007.py
# 1929 소수 구하기
import math
import sys
input = sys.stdin.readline

def isPrim(num):
    if num == 1:
        return False
    
    for i in range(2, int(math.sqrt(num)) + 1):
        if num % i == 0:
            return False
    return True

m, n = map(int, input().split())

for i in range(m, n+1):
    if isPrim(i):
        print(i)